apps/11gist-blog/script.js

```diff
@@ -1,5 +1,136 @@
+            $('ul#posts a').click(function(event) {
+
+                var $thatId = $(this).data('url').id;
+                console.log($thatId);
+
+                // prevent page from reloading
+                event.preventDefault();
+
+                // // when a post is selected, display the #new-comment entry form
+                // $('form#new-comment').css('display', 'block');
+
+                // remove active class from all #posts list-links
+                $('ul#posts a').removeClass('active');
+
+                // add active class to clicked #posts list-link
+                $(this).addClass('active');
+
+                // make ajax request for the post object with this link's data-url key
+                $.ajax($(this).data('url'), {
+
+                    // if request is successful run this new post function with returned post as param
+                    success: function(post) {
+                        var $postCreatedDate = moment(post.created_at).format("dddd, MMMM Do YYYY, h:mm:ss a");
+                        var $postUpdatedDate = moment(post.updated_at).format("dddd, MMMM Do YYYY, h:mm:ss a");
+                        var $postDate = $(`<p>Posted: ${$postCreatedDate}</p><p>Last edited: ${$postUpdatedDate}</p>`);
+
+                        $('div#post-date').html($postDate);
+
+                        // pull the content property from the selected post's 'post.md' file and store it in a var
+                        var postContent = post.files['post.md'].content;
+
+                        // clear the #post div then style the postContent with marked() and insert postContent into #post div
+                        $('div#post').html(marked(postContent));
+
+                        // set #post p to text-align: left
+                        $('div#post p').attr('style', 'text-align: left');
+                    }
+                });
+
+                // make ajax request for the comments object with this link's data-comments_url key
+                $.ajax($(this).data('comments'), {
+
+                    // if request is successful run this new comments function with returned comments array as param
+                    success: function(comments) {
+
+                        // reset the comments list before we loop through and generate each comment
+                        $('ul#comments').empty();
+
+                        // loop through comment object in array and enter each as the param for this generate comment function
+                        comments.forEach(function(comment) {
+
+                            // make an html-encapsulated string that displays and styles new comment
+                            var $comment = $(`<blockquote><li><span class="comment-user">${comment.user.login}</span> says "<span class="comment-body">${comment.body}"</span></li><li><span class="comment-date"> ${moment(comment.updated_at).format("MMMM Do, h:mm:ss a")}</span></li></blockquote>`);
```

Comment:
Nice! You could probably split this up into multiple lines though. Sometimes there are style guides that advocate for < 100 characters per line!